src/main/java/com/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,148 @@
+
+        Order newOrder = new Order();
+        newOrder.setUser(user);
+        newOrder.setStatus(Order.Status.PENDING);
+        newOrder.setTotal(total);
+        newOrder.setOrderDate(LocalDateTime.now());
+        newOrder.setShippingAddress(shippingAddress);
+        Set<OrderItem> orderItems = shoppingCart.getCartItems().stream()
+                .map(item -> orderItemMapper.toOrderItem(item, newOrder))
+                .collect(Collectors.toSet());
+        newOrder.setOrderItems(orderItems);
+
+        Order savedOrder = orderRepository.save(newOrder);
+        shoppingCartRepository.delete(shoppingCart);
+
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    @Transactional
+    public List<OrderResponseDto> findOrderHistory() {
+        return orderRepository.findAllByUser(getCurrentUser()).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional
+    public OrderResponseDto updateOrderStatus(
+            Long orderId, UpdateOrderStatusRequestDto updateOrderStatusRequestDto) {
+        String statusString = updateOrderStatusRequestDto.status();
+
+        Order order = orderRepository.getReferenceById(orderId);
+
+        boolean isValidStatus = Arrays.stream(Order.Status.values())
+                .anyMatch(status -> status.name().equalsIgnoreCase(statusString));
+
+        if (!isValidStatus) {
+            throw new IllegalArgumentException("Order status doesn't exists: " + statusString);
+        }
+
+        order.setStatus(Order.Status.valueOf(statusString));
+        Order savedOrder = orderRepository.save(order);
+
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    @Transactional
+    public List<OrderItemDto> findAllFromOrder(Long orderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() ->
+                        new EntityNotFoundException(ORDER_NOT_FOUND_ERR + orderId));
+
+        return order.getOrderItems().stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional
+    public OrderItemDto findItemFromOrder(Long orderId, Long itemId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() ->
+                        new EntityNotFoundException(ORDER_NOT_FOUND_ERR + orderId));
+
+        OrderItem orderItem = order.getOrderItems().stream()
+                .filter(item -> item.getId().equals(itemId))
+                .findFirst()
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Couldn't find item with ID: " + itemId));
+
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    private User getCurrentUser() {
+        return (User) SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getPrincipal();
+    }
```

Comment:
It's better to getCurrentUser() on Controller layer and then pass userId to Service layer.